backend/src/controllers/report.js

```diff
@@ -183,4 +184,44 @@ controller.getProjectStatusSummaryReportOnRequest = async (request, reply) => {
   }
 };
 
+/**
+ * Get a Project Status Summary Report for a specific project.
+ *
+ * @param   {FastifyRequest} request FastifyRequest is an instance of the standard http or http2 request objects.
+ * @param   {FastifyReply}   reply   FastifyReply is an instance of the standard http or http2 reply types.
+ * @returns {object}
+ */
+controller.getProjectQuarterlyBillingReportOnRequest = async (request, reply) => {
+  controller.userRequires(request, what, "reports_read_all");
+  try {
+    const projectId = Number(request.params.id);
+    const fiscal = Number(request.query.fiscal);
+    const quarter = Number(request.query.quarter);
+    const reportDate = new Date();
+    // Get the data from the database.
+    const result = {
+      project: await model.getProjectById(projectId),
+      deliverables: await model.getDeliverableBudgets(projectId, fiscal, quarter),
+      jv: await model.getJournalVoucher(projectId, fiscal, quarter),
+      client: await model.getClientCoding(projectId),
+      quarter: "Q" + quarter,
+      reportDate: reportDate,
+    };
+    // Calculate grand total from each deliverable amount.
+    result.deliverables_total = result.deliverables.reduce((acc, d) => acc + d.amount, 0);
```

Comment:
not a big fan of 1 letter variables, makes reading code much harder.

Reply:
I figured since it's just a reduce function to calculate the sum and it has a comment explaining it it would be fine, but I agree in general one letter variables are bad.